lib/util.h

```diff
@@ -319,80 +307,14 @@ extern int64_t now_ms(void);
 
 extern clock_t sclock(void);
 
-#define BUF_MMAP    (1<<1)
-
-struct buf {
-    char *s;
-    size_t len;
-    size_t alloc;
-    unsigned flags;
-};
-#define BUF_INITIALIZER { NULL, 0, 0, 0 }
-
-#define buf_new() ((struct buf *) xzmalloc(sizeof(struct buf)))
-#define buf_destroy(b) do { buf_free((b)); free((b)); } while (0)
-#define buf_ensure(b, n) do { if ((b)->alloc < (b)->len + (n)) _buf_ensure((b), (n)); } while (0)
-#define buf_putc(b, c) do { buf_ensure((b), 1); (b)->s[(b)->len++] = (c); } while (0)
-
-void _buf_ensure(struct buf *buf, size_t len);
-const char *buf_cstring(const struct buf *buf);
-const char *buf_cstringnull(const struct buf *buf);
-const char *buf_cstringnull_ifempty(const struct buf *buf);
-const char *buf_cstring_or_empty(const struct buf *buf);
-char *buf_newcstring(struct buf *buf);
-char *buf_release(struct buf *buf);
-char *buf_releasenull(struct buf *buf);
-void buf_getmap(struct buf *buf, const char **base, size_t *len);
-int buf_getline(struct buf *buf, FILE *fp);
-size_t buf_len(const struct buf *buf);
-const char *buf_base(const struct buf *buf);
-void buf_reset(struct buf *buf);
-void buf_truncate(struct buf *buf, ssize_t len);
-void buf_setcstr(struct buf *buf, const char *str);
-void buf_setmap(struct buf *buf, const char *base, size_t len);
-void buf_copy(struct buf *dst, const struct buf *src);
-void buf_append(struct buf *dst, const struct buf *src);
-void buf_appendcstr(struct buf *buf, const char *str);
-void buf_appendoverlap(struct buf *buf, const char *str);
 void buf_appendbit32(struct buf *buf, bit32 num);
```

Comment:
I think the appendbit32/64 should go to `buf.h` as well. This could be accomplished by changing their argument types to `uint32_t` and `uint64_t`. I understand that for `bit64` this will require to decide where to define the shared type definition. As an alternative, I wonder if we could just get rid of the `bit32` and `bit64` type aliases. I never found them particularly useful and I always have to look up their definition to understand their signedness.

Reply:
I think the bit32 and bit64 typedefs were useful before C99 standardised the fixed width types, but I don't think they're so useful anymore.  It wouldn't bother me if we dropped them and just started using uint32_t and uint64_t directly.  But it would be a pretty intrusive find-and-replace change, which I think should be in a standalone PR for ease of code review.

If we assume we'll be doing that at some point, then right now we could either:

1. move the buf_appendbit functions to buf.h, changing their argument types to uint32_t and uint64_t to make it work; or
2. leave them in util.h for now, and move them as part of the greater bit32 change

Do you have an opinion?  I think I prefer 1, but I could go either way.